src/components/DemoSection.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <div class="s-demo-section">
+    <Accordion openedTitle="Hide Code" closedTitle="Show Code">
+      <div slot="content">
+        <pre><code v-html="escapedHtml"></code></pre>
+      </div>
+    </Accordion>
+
+    <div class="s-demo-section__content">
+      <slot name="components"></slot>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Watch, Prop, Vue } from "vue-property-decorator";
+import Accordion from "./Accordion.vue";
+import { escape } from "lodash-es";
+import { VNode } from "vue";
+
+@Component({
+  components: {
+    Accordion
+  }
+})
+export default class DemoSection extends Vue {
+  @Prop()
+  title!: string;
+
+  @Prop({ required: true })
+  code!: string;
+
+  get escapedHtml() {
+    const codeRegEx = new RegExp(
+      `title="${
+        this.title
+      }" :code="demoCode">\\s*<template #components>(.*?)<\\/template>`,
+      "gsm"
+    );
+
+    const codeMatch = codeRegEx.exec(this.code) as string[];
```

Comment:
@joshua-larks have never seen anything like this before - how does `as string[]` work?? (just out of curiosity)

same with `let indentedLines: string[] = [];`

Reply:
Oh its typescript. Just setting the type to an array of strings